src/http/endpoint_key.rs

```diff
@@ -0,0 +1,196 @@
+use std::collections::hash_map::DefaultHasher;
+use std::hash::{Hash, Hasher};
+use std::ops::Deref;
+
+#[derive(Debug)]
+pub struct EndpointKey(reqwest::Request, Vec<String>);
```

Comment:
IT should not be possible to create and EndpointKey for anything but a GET request.